app.js

```diff
@@ -1,49 +1,57 @@
-const express = require('express');
+const express = require("express");
 const app = express();
-const { getTopics, getArticles, getArticleComments, getArticleById } = require('./controller')
-
+const {
+  getTopics,
+  getArticles,
+  getArticleComments,
+  getArticleById,
+  postComment,
+} = require("./controller");
 
 app.use(express.json());
 
-app.get('/api/topics', getTopics);
+app.get("/api/topics", getTopics);
+
+app.get("/api/articles", getArticles);
 
-app.get('/api/articles', getArticles);
+app.get("/api/articles/:article_id/comments", getArticleComments);
 
-app.get('/api/articles/:article_id/comments', getArticleComments);
-app.get('/api/articles/:article_id', getArticleById)
+app.get("/api/articles/:article_id", getArticleById);
 
-app.all('*', (request, response, next) => {
-    response.status(404).send({msg: 'Not found!'});
+app.post("/api/articles/:article_id/comments", postComment);
+
+app.all("*", (request, response, next) => {
+  response.status(404).send({ msg: "Not found!" });
 });
 // app.use((err, req, res, next) => {
 //     if (err.status === 400) {
 //         res.status(400).send({ msg: "Bad request!" });
-//     } else if (err.status === 404) { 
+//     } else if (err.status === 404) {
 //         res.status(404).send({ msg: "Not found!"});
 //     } else {
 //         next(err)
 //     }
 // });
 app.use((err, req, res, next) => {
-    if (err.status && err.msg) {
-        res.status(err.status).send({msg : err.msg})
-    } else {
-        next(err)
-    }
-    })
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
 
 app.use((err, req, res, next) => {
-    if (err.code === '22P02') {
-        res.status(400).send({msg : "Bad request!"})
-    } else {
-        next(err)
-    }
-})
-
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request!" });
+  } else if (err.code === "23502") {
+    res.status(404).send({ msg: "Not found!" });
+  } else {
+    next(err);
+  }
+});
 
 app.use((err, req, res, next) => {
-        res.status(500).send({ msg: "Internal Server Error"});
+  res.status(500).send({ msg: "Internal Server Error" });
 });
 
```

Comment:
Have a think about extracting your errors into a separate file 😄  (I'm pretty sure this'll be covered in the seminars at 9.30!)